twingate/client_group.go

```diff
@@ -210,3 +230,65 @@ func (client *Client) deleteGroup(ctx context.Context, groupID graphql.ID) error
 
 	return nil
 }
+
+type GroupsFilter struct {
+	Name     *string
+	Type     *string
+	IsActive *bool
+}
+
+func (f GroupsFilter) Match(group *Group) bool {
+	if f.Type != nil && *f.Type != string(group.Type) {
+		return false
+	}
+
+	if f.IsActive != nil && *f.IsActive != bool(group.IsActive) {
+		return false
+	}
+
+	return true
+}
+
+func (client *Client) filterGroups(ctx context.Context, filter *GroupsFilter) ([]*Group, error) {
+	if filter == nil {
```

Comment:
If `buildFilter()` always returned a valid `GroupsFilter` object (even if all the attributes are `nil`), you could remove this whole block to simplify the code a bit.  You're already optimizing below which query you want to make to the API based on the individual filters, so this block is essentially a duplicate of all that.

Reply:
thank you, good point

Reply:
fixed